js/compatibility_tests/v3.0.0/test.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+
+set -e
+
+# Download protoc 3.0.0 from Maven if it is not already present.
+OLD_PROTOC_URL=https://repo1.maven.org/maven2/com/google/protobuf/protoc/3.0.0/protoc-3.0.0-linux-x86_64.exe
+if [ ! -f protoc ]; then
+  wget $OLD_PROTOC_URL -O protoc
+  chmod +x protoc
+fi
+
+pushd ../..
+npm install && npm test
+popd
+
+old_protoc=./protoc
+new_protoc=../../../src/protoc
+
+# The protos in group 2 have some dependencies on protos in group 1. The tests
+# will verify that the generated code for one group can be regenerated
+# independently of the other group in a compatible way.
+group1_protos="data.proto test3.proto test5.proto commonjs/test6/test6.proto testbinary.proto testempty.proto test.proto"
```

Comment:
Good question and sorry for the delayed reply. I thought about this a bit and I am thinking that it probably does have to be duplicated actually, because over time we might want to add new test .proto files (or even remove old ones) and thus we can't in general assume that the .proto files used for a particular version's compatibility test match the ones used for the current version. Does that seem fair? On the other hand I guess we could try to factor out some of this logic from gulpfile.js so that for future versions we have a more streamlined way of generating the compatibility test directory (ideally we could have it entirely scripted).